connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/RequestEvents.scala

```diff
@@ -0,0 +1,327 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connect.service
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.google.protobuf.Message
+
+import org.apache.spark.connect.proto
+import org.apache.spark.scheduler.{LiveListenerBus, SparkListenerEvent}
+import org.apache.spark.sql.catalyst.{QueryPlanningTracker, QueryPlanningTrackerCallback}
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.connect.common.ProtoUtils
+import org.apache.spark.util.{Clock, Utils}
+
+object RequestEvents {
+  // TODO: Make this configurable
+  val MAX_STATEMENT_TEXT_SIZE = 65535
+}
+
+/**
+ * Post request Connect events to @link org.apache.spark.scheduler.LiveListenerBus.
+ *
+ * @param planHolder:
+ *   Request for which the events are generated.
+ * @param clock:
+ *   Source of time for unit tests.
+ */
+case class RequestEvents(planHolder: ExecutePlanHolder, clock: Clock)
+    extends QueryPlanningTrackerCallback {
+
+  /**
+   * Post @link org.apache.spark.sql.connect.service.SparkListenerConnectOperationStarted.
+   */
+  def postStarted(): Unit = {
+    val request = planHolder.request
+    val plan: Message =
+      request.getPlan.getOpTypeCase match {
+        case proto.Plan.OpTypeCase.COMMAND => request.getPlan.getCommand
+        case proto.Plan.OpTypeCase.ROOT => request.getPlan.getRoot
+        case _ =>
+          throw new UnsupportedOperationException(
+            s"${request.getPlan.getOpTypeCase} not supported.")
+      }
```

Comment:
not used?

Reply:
This is used to generate the [statementText](https://github.com/apache/spark/pull/41443/files#diff-ade438a128aec58850ce538b342097098ce2df090d89a1b0ca67b495a407e094R68)

Reply:
I'm blind :)